internal/plugin/grpc_client_plugin.go

```diff
@@ -79,14 +84,72 @@ func (gc *GrpcClient) Init(ctx context.Context, messagePipe bus.MessagePipeInter
 	gc.conn, err = grpc.DialContext(grpcClientCtx, serverAddr, agentGrpc.GetDialOptions(gc.config)...)
 	gc.connectionMutex.Unlock()
 
+	gc.fileServiceClient = v1.NewFileServiceClient(gc.conn)
+
 	if err != nil {
 		return err
 	}
 	backOffCtx, backoffCancel := context.WithTimeout(ctx, gc.config.Client.Timeout)
 
+	err = backoff.WaitUntil(backOffCtx, gc.config.Common, gc.createConnectionClient)
 	defer backoffCancel()
+	if err != nil {
+		return err
+	}
+
+	subscribeCtx, gc.subscribeCancel = context.WithCancel(ctx)
+
+	go gc.subscribe(subscribeCtx)
 
-	return backoff.WaitUntil(backOffCtx, gc.config.Common, gc.createConnectionClient)
+	return nil
+}
+
+// wastedassign giving out that subscribeClient is set to nil
+// nolint: wastedassign
+func (gc *GrpcClient) subscribe(ctx context.Context) {
+	var subscribeClient v1.CommandService_SubscribeClient
+	var err error
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			subscribeClient, err = gc.commandServiceClient.Subscribe(ctx)
+			if err != nil {
+				slog.ErrorContext(ctx, "error subscribing: ", err)
```

Comment:
```suggestion
				slog.ErrorContext(ctx, "Unable subscribe" "error", err)
```

Reply:
I would even consider adding the extra "to" 😆 
```suggestion
				slog.ErrorContext(ctx, "Unable to subscribe" "error", err)
```